src/main/java/com/bandwidth/sqs/queue/buffer/task_buffer/KeyedTaskBuffer.java

```diff
@@ -11,11 +11,20 @@
 import java.util.Timer;
 import java.util.TimerTask;
 
+import io.reactivex.subjects.SingleSubject;
+
+/**
+ * A buffer that buffers individual data, collecting it in bucket by key, then running a task to batch
+ * process the data
+ *
+ * @param <K> Key
+ * @param <D> Data
+ */
 public class KeyedTaskBuffer<K, D> {
     private Timer timer = new Timer();
     private final int maxBufferSize;
     private final Duration maxWait;
-    private final Map<K, Buffer<D>> buffers = new HashMap<>();
+    private final Map<K, List<D>> buffers = new HashMap<>();
```

Comment:
Not sure if that's the same abstraction I'm using it as. Multimap is really a 1 to many map, where mine is a 1-1 map where the value happens to be a list. I don't think any of the features Multimap adds would really help here.